### PR TITLE
Make PressableWithSecondaryInteraction into a wrapper instead of a HOC

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
This component should really have been a wrapper and not a HOC.

### Fixed Issues (partial) 
Related to https://github.com/Expensify/Expensify/issues/147472

### Tests
1) Change `ReportActionItem` like so:

```jsx
<PressableWithSecondaryInteraction
    onSecondaryInteraction={() => console.log('secondary interaction!')}
    onPress={() => console.log('regular press!')}
>
    {!props.displayAsGroup
        ? <ReportActionItemSingle action={props.action} />
        : <ReportActionItemGrouped action={props.action} />}
</PressableWithSecondaryInteraction>
```
2) On web/desktop:
    a) right-click on a report action, verify that you see 'secondary interaction!' in the logs. Also verify that the default context menu does not appear.
    b) longpress on a report action, verify that you see 'secondary interaction!' in the logs.
    c) left-click on a report action, verify that you see 'regular press!' in the logs.
3) On iOS/Android/mWeb:
    a) longpress on a report action, verify that you see 'secondary interaction!' in the logs. ~~Also verify that the default context menu does not appear (option to copy on iOS)~~. (this last one didn't have a simple solution, so I'd rather wait to see if it's a real problem).
    b) do a regular tap on a report action, verify that you see 'regular press!' in the logs.
    

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
image

#### Mobile Web
image

#### iOS
image

### CODE REVIEW

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (alex-mechler) [inline src/components/PressableWithSecondaryInteraction/index.js:2]: I _think_ we actually prefer these right above the class declaration

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (alex-mechler) [inline src/components/PressableWithSecondaryInteraction/index.js:10]: Comments on the propTypes 😄

File: src/components/PressableWithSecondaryInteraction/index.native.js
Comment (alex-mechler) [inline src/components/PressableWithSecondaryInteraction/index.native.js:2]: Same thing about comments in this file

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js]: +1 on having just imports at the very top. Not sure there's any specific reason. Consistency?

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [inline src/components/PressableWithSecondaryInteraction/index.js:20]: Last time I checked we are not using `React.createRef()`. I am generally cool with this since `createRef()` adds the `current` and it takes an extra second to think about which one you are dealing with.

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [inline src/components/PressableWithSecondaryInteraction/index.js:24]: Whenever you add a listener in `componentDidMount()` you should clean it up by removing in `componentWillUnmount()`. Gonna guess there is no `onContextMenu` prop?

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [inline src/components/PressableWithSecondaryInteraction/index.js:31]: you'll probably catch this but I don't think you want to remove `children` or `onSecondaryInteraction`.

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js:31]: Oh I see you're not overwriting props...

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.js:24]: Good point! Yep, no `onContextMenu` prop

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js:31]: Hmm I think you can just spread all the props in and then overwrite `onLongPress`

Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.js:30]: But wouldn't you also want to prevent passing the children in? IDK, might not actually do anything but the children shouldn't be passed as props to the `Pressable` right?

**Review by alex-mechler — DISMISSED** [review]
Looking good to me and tests well.

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js:30]: Is there a difference between

```jsx
<Something>
    {this.props.children}
</Something>
```
and 

```js
<Something children={this.props.children}/>
```

File: src/components/PressableWithSecondaryInteraction/index.native.js
Comment (marcaaron) [inline src/components/PressableWithSecondaryInteraction/index.native.js:31]: Add a `displayName` to this?

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js:30]: downgrading this one to a NAB

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/components/PressableWithSecondaryInteraction/index.native.js
Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: Doesn't look like that's necessary for a wrapper component:

image

Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: Interesting. It's a stateless component and we usually add the `displayName`.

Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: Why do we not need it in this case?

Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: The plot thickens! Adding `dispalyName` on stateless components is not something we've been consistent about in this codebase so far. Some examples of stateless components without `displayName` are:

1) `src/components/Avatar.js`
2) `src/pages/home/report/ReportActionItem.js`
3) `src/components/TextInputWithFocusStyles.js`
4) `src/components/ThumbnailImage.js`

There are probably more examples because I didn't go look through every file in the repo. Anyways, I'm happy to add the display name here because it doesn't do any harm, but this inconsistency is something we should make a decision on and address at some point.

Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: Interesting, not 100% sure but I _think_ I figured out the story here. If a stateless component uses a regular function (not an arrow function), then it will use the function's name as the display name. If it uses an arrow function, you need to provide a display name. However, all our code (native and web/desktop I believe), goes through a babel pre-processor which converts all ES6 arrow functions into regular named functions. So the display name isn't necessary anymore.

Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: That also explains why it was never necessary for class components.

Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: >Adding dispalyName on stateless components is not something we've been consistent about in this codebase so far

Consistency will never be perfect and always will be us doing our best to make things consistent. Let's do what we can and if we spot inconsistency we can address it in reviews or create issues to clean things up.

>goes through a babel pre-processor which converts all ES6 arrow functions into regular named functions

Yes, but what we see on dev is not always what we see on production.

**Dev**
Screen Shot 2021-01-25 at 9 19 29 AM

**Production**
Screen Shot 2021-01-25 at 9 17 30 AM

Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: Maybe we should also add something to `STYLE.md` about why the `displayName` is important now that we know the reason.

Comment (roryabraham) [reply src/components/PressableWithSecondaryInteraction/index.native.js:31]: Cool. Added the display name 👍

**Review by alex-mechler — DISMISSED** [review]

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [inline src/components/PressableWithSecondaryInteraction/index.js:33]: NAB, should this be `ContextMenuEvent`? Not sure if it's the case...

Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js:33]: oh actually I think contextmenu is a kind of `MouseEvent`

Comment (marcaaron) [reply src/components/PressableWithSecondaryInteraction/index.js:33]: Screen Shot 2021-01-25 at 12 14 52 PM

**Review by marcaaron — DISMISSED** [review]
LGTM just left question about JSDoc but won't block on it,

**Comment by alex-mechler** [comment]
Got conflicts now

**Review by Dal-Papa — DISMISSED** [review]
Not particularly proficient with the code but it LGTM after fixing conflicts.

**Comment by roryabraham** [comment]
Merge conflict resolved 👍

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (marcaaron) [inline src/components/PressableWithSecondaryInteraction/index.js:33]: Type is a `@param {MouseEvent}`

**Review by marcaaron — APPROVED** [review]

**Pull request merged**
